interactive/plots/flow_diagram.py

```diff
@@ -1,798 +1,309 @@
-                    and tmp >= userMM[0]):
-                if reactant in list_of_species:
-                    edgeColors.update({name: "#6b6bdb"})
-                else:
-                    edgeColors.update({name: "#94b8f8"})
-            else:
-                if tmp > userMM[1]:
-                    tmp = userMM[1]
-                elif tmp < userMM[0]:
-                    tmp = userMM[0]
-                edgeColors.update({name: "#e0e0e0"})
-            if (reactant not in blockedSpecies
-                    and tmp_reaction not in blockedSpecies):
-                raw_yields.update(
-                    {reactant+"__TO__"+reaction: tmp})
-    return (raw_yields, edgeColors, quantities,
-            total_quantites, reaction_names_on_hover)
-
-
-# create line widths scaling based on user selected option
-def calculateLineWeights(maxWidth, species_yields, scale_type):
-    # use fluxes from map we setup, scale the values, add weights and whatnot
-    maxVal = -1
-    minVal = 99999999
-    rawVal = {}
-    numOfInvalids = 0
-    if scale_type == 'log':
-        if species_yields != {}:
-
-            li = species_yields
-            logged = []
-            for i in li:
-                # fail safe for null->null value
-                if (str(i) != 'null->null'
-                        and float(0.0) != float(species_yields[i])):
-                    rawVal.update({i: species_yields[i]})
-                    try:
-                        logged.append((i, math.log(species_yields[i])))
-                        if species_yields[i] < minVal:
-                            minVal = species_yields[i]
-                        if species_yields[i] > maxVal:
-                            maxVal = species_yields[i]
-                    except ValueError:
-                        numOfInvalids = numOfInvalids+1
-                        logged.append(
-                            (i, abs(math.log(abs(species_yields[i])))))
-                        if abs(species_yields[i]) < minVal:
-                            minVal = abs(species_yields[i])
-                        if abs(species_yields[i]) > maxVal:
-                            maxVal = abs(species_yields[i])
-
-            vals = [i[1] for i in logged]
-            min_val = abs(min(vals, default="EMPTY"))
-            range = max(vals) - min(vals, default="EMPTY")
-            if max(vals) == min(vals, default="EMPTY"):
-                range = 1
-            scaled = [(x[0], (float(((x[1] + min_val)/range))
-                       * float(maxWidth)) + 1) for x in logged]
-            return list(scaled), minVal, maxVal, rawVal
-        else:
-            return list([]), -1, 999999999, {}
+def np_encoder(object):
+    if isinstance(object, np.generic):
+        return object.item()
+
+
+# time interval should be in format [small, large]
+# and have a minimum difference greater than or equal to the model timestep
+def return_valid_df_time_indices(df, user_time_range, model_timestep):
```

Comment:
Please convert all of these comments into [docstrings](https://peps.python.org/pep-0257/#multi-line-docstrings)